django/db/models/expressions.py

```diff
@@ -507,6 +507,15 @@ def as_sql(self, compiler, connection, function=None, template=None):
         template = template or self.extra.get('template', self.template)
         return template % self.extra, params
 
+    def as_sqlite(self, *args, **kwargs):
+        sql, params = self.as_sql(*args, **kwargs)
+        try:
```

Comment:
I think this is fine as it is. I don't like that we have to check for the exception or fallback to using the private output_field_or_none though. Problem for another day.